### ISSUE TITLE
False positive with key sequence such as "\C-c\C-ia"

### ISSUE THREAD

**Issue by kobarity** [purcell/package-lint#239]
Key sequences such as "\C-c\C-ia" are reported as an error. This can be tested as follows:

```elisp
(package-lint-test--run "(define-key map \"\C-c\C-ia\" #'something)")
```

This seems to happen because `read-kbd-macro` (and its callees `kbd` and `key-parse`) used in `package-lint--extract-key-sequence` skips "\C-i".
